### PR TITLE
Euler1

### PR DESCRIPTION
**Pull request by Ceritoxi**

### CODE REVIEW

File: euler-maven-1/src/main/java/com/ceritoxi/Euler1.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	
+	private int sum;
+	
+	public Euler1() {
+		this.sum = 0;
+		task();
+	}
+	
+	private void task() {
+		this.sum = Math.sumOfNumbersUnderZThatAreDividableByXorY(3, 5, 1000);
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Euler1.java:13]: this is a tad overkill, you can just make the call to the Math class at the constructor in line 8.

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.ceritoxi;
+
+public class Math {
+	
+	public static boolean isDividableByXorY(int dividend, int numberX, int numberY) {
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Math.java:5]: numberX and numberY -> the word "number" adds no value. You can use x and y instead. Its safe in these low level calculation cases

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		if (numberX == 0 && numberY == 0) return false ;
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Math.java:6]: Always use code block separators (curly) even if there is just one line in it such as:
```
if (numberX == 0 && numberY == 0) {
 return false ;
}
```

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		if (numberX != 0 && dividend % numberX == 0) return true;
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Math.java:7]: the  zero check already happened in the previous line. Add further logic in an "else if" clause, also its redundant to use two if, you can do it in one go. such as:
```
} else if ( (dividend % numberX == 0) || (dividend % numberY == 0)) {
 return true;
}

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+		if (numberY != 0 && dividend % numberY == 0) return true;
+	
+		return false;
+	}
+	
+	public static int sumOfNumbersUnderZThatAreDividableByXorY (int numberX, int numberY, int numberZ) {
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Math.java:13]: Just as before use x ,y, z as names

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+		int sum = 0;
+		
+		if (numberZ < 1) return sum;
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Math.java:16]: curly missing

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+			
+		for (int i = 1; i < numberZ; i++) {
+			if (isDividableByXorY(i, numberX, numberY)) sum += i;
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Math.java:19]: curly missing

File: euler-maven-1/src/main/java/com/ceritoxi/Runner.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.ceritoxi;
+
+public class Runner 
+{
+    public static void main(String[] borsodi) {
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Runner.java:5]: nofun

while I get the joke, later on nobody will know wtf is borsodi supposed to mean :D. Use args or arguments plz.

File: euler-maven-1/src/main/java/com/ceritoxi/Runner.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+		Euler1 run = new Euler1();
+		run.printResult();
```
Comment (dkartacs) [inline euler-maven-1/src/main/java/com/ceritoxi/Runner.java:7]: naming would make more sense the other was around:
```
Euler1 euler1 = new Euler1();
euler1.run();
```

File: euler-maven-1/src/test/java/com/ceritoxi/MathTest.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+		assertEquals(true, Math.isDividableByXorY(100, 3,  5));
+		assertEquals(true, Math.isDividableByXorY(100, 5,  3));
+		assertEquals(true, Math.isDividableByXorY(100, 2,  5));
+		assertEquals(true, Math.isDividableByXorY(-100, 2,  5));
+		assertEquals(true, Math.isDividableByXorY(-100, 2,  5));
+		assertEquals(true, Math.isDividableByXorY(-100, 2,  5));
+    }
+	
+	public void testMathSumOfNumbersUnderZThatAreDividableByXorY() {
+		//if (Math.sumOfNumbersUnderZThatAreDividableByXorY(x,y,z))
```
Comment (dkartacs) [inline euler-maven-1/src/test/java/com/ceritoxi/MathTest.java:34]: remove comment

**Review by dkartacs — CHANGES_REQUESTED** [review]
Added a few comment, plz change these (missing curly, "number" in variable name, etc...) in the other PR-s as well, Let me know when its done and I will continue

File: euler-maven-1/src/main/java/com/ceritoxi/Math.java (hunk 4 above)
Comment (Ceritoxi) [reply euler-maven-1/src/main/java/com/ceritoxi/Math.java:7]: Well the code that you wrote is flawed, but indeed the first line of my code if completely useless now that I think about it. So I will simply remove it. With that change the else if is no longer needed.

**Pull request closed without merging**